### ISSUE TITLE
Create noop resource

### ISSUE THREAD

**Issue by piavlo** [hashicorp/terraform#580]
I need a kind of meta resource that creates nothing but can specify depends_on and invoke provisioners

for example i want to create 3 instances (on which some clustered service will be deployed) but all instances need to be in different zones, so i cannot use count=3 in one resource, i need to run a provisioner only after all 3 instances have been created

**Comment by phinze** [comment]
In the absence of a more first-class concept like what's being discussed here, I've personally found `null_resource` to be an absolutely invaluable--if a bit inelegant--tool in setting up complex systems. I think it's probably worth documenting it as-is until we gain something better.

**Comment by phinze** [comment]
More valuable thoughts over in #1181, which I just closed to merge the discussion back here.

**Comment by apparentlymart** [comment]
Over in #2696 I proposed a patch that adds a resource that behaves quite a lot like ``null_resource`` except that it has a ``state_key`` property that allows the user to describe situations where the provisioners need to be re-run for some reason.

This is aimed at the same use-case as described in this issue: an "escape hatch" to run provisioners without a "real" resource to attach them to, but with the extra possibility of re-running those provisioners when the world changes around them. For example, I have prototyped using it to re-run some installation steps after pushing a new code bundle up to S3, with the S3 key (which contains a version number and build timestamp) inserted into the ``state_key``.
